pkg/uhttp/wrapper.go

```diff
@@ -254,6 +289,12 @@ func (c *BaseHttpClient) Do(req *http.Request, options ...DoOption) (*http.Respo
 		resp     *http.Response
 	)
 	l := ctxzap.Extract(req.Context())
+
+	// If a rate limiter is defined, take a token before making the request.
+	if c.rateLimiter != nil {
+		c.rateLimiter.Take()
```

Comment:
Does taking a token just block until one is available?

Reply:
Yes